### ISSUE TITLE
Allowing serverless to run in interactive debug mode

### ISSUE THREAD

**Issue by Bilal-S** [serverless/serverless#1724]
Feature Request Template ------------------------------------------
##### Feature Request:

Interactive Debugger for Serverless
I am expanding this from issue #1720 
The need existed for me to understand more clearly how the framework operates. An interactive debugging connection seems the solution for me. I will add the code for review as pull request.
##### Benefits:
- interactive debugging
- set breakpoints, trace call history, call stack, interactive command window, etc.
- tremendous speed up for people contributing to the platform
##### Drawbacks:
- There does not seem to be slick way of exiting the debugging session. The V8 profiler window in browser will need to be closed by developer.
- pre-requisites exist: installation of node-inspector package and chrome browser
##### Additional Details:

Discussion template  ----------------------------------------------
##### TLDR:

Adding connection to node-inspector so interactive debugging can occur

**Comment by Bilal-S** [comment]
I seem to be off-track with the Alpha release changes. I've been changing things in older code. I will review the latest version. Is v1 the latest branch?

**Comment by flomotlik** [comment]
@Bilal-S we just moved from v1.0 to the master branch, so everything can run against the master now.

**Comment by flomotlik** [comment]
We're going to look into debugging with #2609 so I'm closing this here.

**Comment by bencooling** [comment]
I think #2609 and it's child #2722 issue is about performance profiling not stepping through code with interactive debugging. Is there an issue for this topic?

**Comment by pmuens** [comment]
@bencooling yes, you're right. Those two issues are about profiling / making Serverless faster.

AFAIK we don't have another issue for interactive debugging (except this one).

+1 for not introducing Node inspector and relying on native debugging. Do you have any idea how this might look like in day-to-day usage / how we should implement debugging?

**Comment by pmuens** [comment]
Reopening this one as the two other issues discuss profiling...

**Comment by bencooling** [comment]
Sorry I thought about it some more, you can run actually Serverless with the [V8 inspector](https://nodejs.org/api/debugger.html#debugger_v8_inspector_integration_for_node_js) using the following command:

`node --inspect --debug-brk ./node_modules/.bin/serverless`

Of course if serverless is global installed you could replace the local relative path with an absolute path. 

I'm not sure if interactive debugging sls needs to be documented for reference as its not a Serverless specific issue, but for what it is worth I did come looking here for an answer.

**Comment by pmuens** [comment]
@bencooling thanks for looking into it 👍 

I agree. Debugging Serverless with the V8 inspector is not a Serverless related issue.

Closing this for now as you've provided a good solution to resolve the debugging problem and we have no direct plans to work on this anytime soon.
